assets/src/amp-validation/counts/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { isPlainObject } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import domReady from '@wordpress/dom-ready';
+
+/**
+ * Internal dependencies
+ */
+import './style.css';
+
+/**
+ * Updates a menu item with its count.
+ *
+ * If the count is not a number or is `0`, the element that contains the count is instead removed (as it would be no longer relevant).
+ *
+ * @param {HTMLElement} itemEl Menu item element.
+ * @param {number} count Count to set.
+ */
+function updateMenuItem( itemEl, count ) {
+	count = Math.abs( count );
+
+	if ( isNaN( count ) || count === 0 ) {
+		itemEl.parentNode.parentNode.removeChild( itemEl.parentNode );
+	} else {
+		itemEl.classList.remove( 'loading' );
+		itemEl.textContent = count.toLocaleString();
+	}
+}
+
+/**
+ * Updates the 'Validated URLs' and 'Error Index' menu items with their respective unreviewed count.
+ *
+ * @param {any} counts Counts for menu items.
+ * @param {number} counts.validated_urls Unreviewed validated URLs count.
+ * @param {number} counts.errors Unreviewed validation errors count.
+ */
+function updateMenuItemCounts( counts ) {
+	if ( ! isPlainObject( counts ) ) {
+		// eslint-disable-next-line no-console
+		console.error( '[AMP Plugin] An error occurred while retrieving unreviewed validation counts. Received: ' + JSON.stringify( counts ) );
+		counts = {};
+	}
+
+	const { validated_urls: newValidatedUrlCount, errors: newErrorCount } = counts;
+
+	const errorCountEl = document.getElementById( 'new-error-index-count' );
+	const validatedUrlsCountEl = document.getElementById( 'new-validation-url-count' );
+
+	updateMenuItem( errorCountEl, newErrorCount );
+	updateMenuItem( validatedUrlsCountEl, newValidatedUrlCount );
+}
+
+domReady( async () => {
+	const counts = await apiFetch( { path: '/amp/v1/unreviewed-validation-counts' } );
+	updateMenuItemCounts( counts );
```

Comment:
I'm looking at `apiFetch` here: 

https://github.com/WordPress/gutenberg/blob/30b2c144afc8586602c419d2faf21b7f9721be7b/packages/api-fetch/src/index.js#L57-L63

It's throwing an error if the request fails including if a non-successful response code is being sent. Therefore, I think actually `updateMenuItemCounts` can be changed to only accept an `object` and that this code here can be changed to:

```suggestion
	try {
		const counts = await apiFetch( { path: '/amp/v1/unreviewed-validation-counts' } );
		if ( ! isPlainObject( counts ) ) {
			throw new Error( 'Object not returned.' );
		}
		updateMenuItemCounts( counts );
	} catch ( exception ) {
		// eslint-disable-next-line no-console
		console.error( '[AMP Plugin] An error occurred while retrieving unreviewed validation counts.', exception.message );
	}
```

Reply:
Ah, right. I've taken a similar approach to what you've suggested in https://github.com/ampproject/amp-wp/pull/5900/commits/8d90f2d280b8fc3dced914e4f545e1e46182c909.